### ISSUE TITLE
Please remove the dependency on libcsparse.so, because it is a subset of libcxsparse.so, and isn't built any more

### ISSUE THREAD

**Issue by yurivict** [naev/naev#774]
This dependency https://github.com/naev/naev/blob/master/configure.ac#L365 refers to the library that isn't installed by the SuiteSparse project any more.

The upstream author said ```"The default build creates libcxsparse.so, which is a superset of CSparse (using the CXSparse library)."```

**Comment by bobbens** [comment]
Although it should be possible to remove it easily, when not found, it should default to an internal csparse version that is included with naev. When you run ./configure, it should print something like "csparse: internal" indicating that it is being used instead of the library version.

**Comment by yurivict** [comment]
@bobbens Thanks for your suggestion, it works.
csparse lookup should still be removed though.

**Comment by bobbens** [comment]
Yeah, either way, the csparse stuff was implemented but was never really satisfactorily working so it's not actually used. I think it would be probably best to get rid of it, I'll leave this issue open until it's done.

**Comment by onpon4** [comment]
I think this is still unsolved, correct? (Just did a quick grep -r and I'm still seeing references to csparse.)

**Comment by bobbens** [comment]
There is still talk about using linear algebra to do economy stuff and I think it could be used in that case. However, we are packaging it and the check should fail if it's not there, so it shouldn't be a problem. Closing.

**Comment by akien-mga** [comment]
This is still an issue for the reference, I'm running into this problem when packaging Naev for Mageia.

Your configure checks mandate both `csparse` and `cxsparse`, but Mageia only provides `cxsparse` (as it's a superset of `csparse`).

I currently use this hack to work it around:
```diff
Index: naev-0.8.1/configure.ac
===================================================================
--- naev-0.8.1/configure.ac
+++ naev-0.8.1/configure.ac	2021-01-26 11:18:58.321557951 +0100
@@ -290,13 +290,13 @@
 AC_CHECK_LIB([csparse], [cs_spalloc], [have_csparse_lib=yes], [have_csparse_lib=no])
 AC_CHECK_LIB([cxsparse], [cs_di_spalloc], [have_cxsparse_lib=yes], [have_cxsparse_lib=no])
 
-AS_IF([test "x$have_csparse_lib" = "xyes" -a "x$have_cxsparse_lib" = "xyes"], [
+AS_IF([test "x$have_csparse_lib" = "xyes" -o "x$have_cxsparse_lib" = "xyes"], [
     AC_CHECK_HEADERS([cs.h], [have_cs_h=yes], [have_cs_h=no])
     AC_CHECK_HEADERS([suitesparse/cs.h], [have_suitesparse_cs_h=yes], [have_suitesparse_cs_h=no])
 
     AS_IF([test $have_cs_h = yes -o $have_suitesparse_cs_h = yes], [
         used_csparse=shared
-        CSPARSE_LIBS="-lcsparse -lcxsparse"
+        CSPARSE_LIBS="-lcxsparse"
     ], [
         used_csparse=internal
     ])
```

**Comment by UncombedCoconut** [comment]
I've pushed a less hacky fix to the 0.8 branch.
Just to set expectations: it will only see the light of day if there's a compelling reason to release Naev 0.8.2. So far the benefits would be minor (German translation coverage, tweaked chances of seeing bar missions, tweaked pirate ship-stealing mission rewards).
Barring larger issues, I think the consensus is to focus on 0.9.0, which fully adopts Meson as the build/install system and never had this problem.
